lib/pharos/addons/kubernetes-dashboard/addon.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+Pharos.addon 'kubernetes-dashboard' do
+  name 'kubernetes-dashboard'
+  version '1.8.3'
+  license 'Apache License 2.0'
+
+  install {
+    install
+
+    logger.info { "~~> kubernetes-dashboard can be accessed via kubectl proxy at http://localhost:8001/ui" }
+
+    service_account = kube_client.get_service_account('dashboard-admin', 'kube-system')
+    token_secret = service_account.secrets[0]
+    return unless token_secret
```

Comment:
Hmm.  I think this (or any other `return` for that matter) will give you a `LocalJumpError`.

```ruby
class Addon
  def self.hooks
    @hooks ||= {}
  end

  def self.install(&block)
    hooks[:install] = block
  end

  def run_install
    puts "Installing.."
    self.class.hooks[:install].call
    puts "Installed!"
  end
end

class FooAddon < Addon
  install do
    return "hello"
  end
end

FooAddon.new.run_install
```

==> 

```console
Installing..
Traceback (most recent call last):
	2: from foo.rb:23:in `<main>'
	1: from foo.rb:12:in `run_install'
foo.rb:19:in `block in <class:FooAddon>': unexpected return (LocalJumpError)
```

Reply:
Does it throw `LocalJumpError` also if `instance_eval` is used (it's using instance_eval currently)?

Reply:
The problem is that a return from a proc returns from the thing that is calling the proc, not from the proc itself. Just like:

```ruby
def run
  foo.each do |xxx|
    return "foo" 
  end
end
```

does not returrn from the `.each` block but the whole `run`.

Reply:
Makes sense, need to fix kubernetes-dashboard addon.

Reply:
Fixed.